Replanetizer/Frames/ModelFrame.cs

```diff
@@ -429,24 +429,49 @@ private void OnResize()
             invalidate = true;
         }
 
-        private void ExportModel(Model model)
+        private void ExportSelectedModel()
         {
-            if (selectedModel == null) return;
+            var model = selectedModel;
+            if (model == null) return;
 
             var fileName = CrossFileDialog.SaveFile(filter: ".obj;.iqe");
-            if (fileName.Length > 0)
+            if (fileName.Length == 0) return;
+
+            var extension = Path.GetExtension(fileName);
+            switch (extension)
             {
-                string extension = Path.GetExtension(fileName);
+                case ".obj":
+                    ModelWriter.WriteObj(fileName, model);
+                    break;
+                case ".iqe":
+                    ModelWriter.WriteIqe(fileName, level, model);
+                    break;
+            }
+        }
 
-                switch (extension)
-                {
-                    case ".obj":
-                        ModelWriter.WriteObj(fileName, selectedModel);
-                        break;
-                    case ".iqe":
-                        ModelWriter.WriteIqe(fileName, level, selectedModel);
-                        break;
-                }
+        private void ExportSelectedModelTextures()
+        {
+            var textures = modelTextureList;
+            var textureConfig = selectedModel?.textureConfig;
+            if (textures == null || textureConfig == null) return;
+
+            if (textures.Count != textureConfig.Count)
+            {
+                Logger.Error(
+                    "textures and textureConfig do not have the same count. " +
+                    "We will use the count of the textures and ignore any " +
+                    "extra textureConfigs."
+                    );
+            }
+
+            var folder = CrossFileDialog.OpenFolder();
+            if (folder.Length == 0) return;
+
+            for (var i = 0; i<textures.Count; i++)
```

Comment:
I think there is an issue if textureConfig.Count != textures.Count since if that is the case then we must have skipped a texture in the UpdateTextures() funtion. However, this means that the i-th textureConfig may not correspond to the i-th texture. 

I think a simple fix would be to store the ID and texture in the modelTextureList and simply use these then for the export instead of relying on the textureConfig data.

Reply:
Okay, I'll investigate where textureConfig is populated. I added this condition because one of the models I exported had a config with an ID of -1, but no associated texture.

Reply:
Got it! I just copied the logic from UpdateTextures into ExportSelectedModelTextures. It now iterates through the TextureConfigs, skips negative or out-of-range IDs, and indexes selectedTextureSet to get the Texture.